### DISCUSSION TITLE
Flux Model support ?

### DISCUSSION THREAD

**Discussion by vicento** [lllyasviel/Fooocus#3440] [Q&A] [answered]
Hi Fooocus team. What about supporting the new flux model from [](https://blackforestlabs.ai) ?

**Comment by mashb1t** [comment, marked as answer]
Duplicate of https://github.com/lllyasviel/Fooocus/discussions/3426#discussioncomment-10225778, please find more information here.

**Comment by CantainPike** [comment]
I tested Flux online from Hugging Face and Replicate.com. Flux's image quality is amazing, and its prompt accuracy is great. I think this model will be the future of AI image generation. Fooocus needs to find a way to implement these new models. There are many emerging models like Pixartsigma, HunYuan DIT, Kolors, Lumina, and more. People will gradually lose interest in SDXL, and Stabitty AI has already lost its golden days; many people no longer like it.

**Reply by IPv6** [reply]
Yes, it is amasing. But the question is it amasing enough for you to spend 3-6 months by rewriting whole project and experimenting with all supplementary stuff 8 hours per day though? :) Because it is too different to SDXL, unfortunately (or fortunately... there is always a places for something new)

**Reply by vicento** [reply]
comfyUI was implemented day one and the first optimisations day 2 ; that doesn't seem completely impossible. (hope)

**Reply by IPv6** [reply]
It is not impossible, but Fooocus is strictly SDXL-centric - unlike ComfyUI which is modular from the ground up, it is the whole idea of throwing nodes :)

**Reply by mirek190** [reply]
foocus is based on comfyiu  ... so

**Reply by mashb1t** [reply]
@mirek190 nothing is holding you back, go ahead, the source code is yours, happy to review your PR!

Keep in mind that Flux needs a whole new set of features to be loaded, e.g. other clip encoding etc., needs ldm_patched to be updated, various internal patches to be rewritten and doesn't support any of Fooocus' features except basic image generation, so LoRAs, inpaint, outpaint, enhance, image prompt, ControlNets etc. all won't work and hardware requirements are high (12-16GB VRAM minimum, better 24GB), so be prepared for harsh community feedback on this one. Looking forward to your contribution ;)

**Reply by Moorer624** [reply]
Keep in mind your competitors are already making use of Flux. Its not going away anytime soon. If I have to switch to ComfyUI to use Flux models, I'll reluctantly do so, but I'd rather keep using Fooocus. Just saying...

**Reply by mashb1t** [reply]
@Moorer624 there is no competition between open-source tools for Stable Diffusion, just a bunch of nerds putting their heart and soul in. Fooocus isn't in active development anymore, so don't expect new features anytime soon.
Feel free to implement Flux support though by contributing via pull request.

**Comment by bsdesigner** [comment]
Maybe we should ask lllyasviel to design a new "Fooocus FLUX", if of course he'd be interested....?

**Reply by mashb1t** [reply]
@bsdesigner Fooocus Flux basically would already be equal to their HF space https://huggingface.co/spaces/black-forest-labs/FLUX.1-dev. You can download the code from there and run it locally. Flux currently doesn't support anything else than shown in the space.

**Reply by bsdesigner** [reply]
Yes, I understand all the problems of today. But I am trying to look into the future (at least a year ahead), as I want to understand the place of our favourite "Fooocus" among the generators of the near future. I think that everyone who had a chance to test the Flux model will confirm how painful it is to go back to SDLX. It's like trying a Mercedes-Benz and then going back to your old, shabby Saturn....

**Reply by mirek190** [reply]
> Yes, I understand all the problems of today. But I am trying to look into the future (at least a year ahead), as I want to understand the place of our favourite "Fooocus" among the generators of the near future. I think that everyone who had a chance to test the Flux model will confirm how painful it is to go back to SDLX. It's like trying a Mercedes-Benz and then going back to your old, shabby Saturn....

Yes sdxl is already outdated from the premiere of FLUX

**Reply by mashb1t** [reply]
@mirek190 can't confirm this in all aspects. Only txt2img is better in Flux, the rest isn't supported and training on distilled data certainly isn't going to provide best results in the future if the community tries to implement new features.

**Reply by bsdesigner** [reply]
> Yes sdxl is already outdated from the premiere of FLUX

It's not really all that clear cut... For example, right now I'm doing sketches for a video about Superagent in the style of Mike Deodato. So, Fooocus understands this artist's art style perfectly and "juggernautXL_v9Rdphoto2Lightning" with "add-detail-xl", "xl_more_art-full" and "MJ52" gives a great result, while "Flux", absolutely does not understand this art style. Here is a typical example of the difference - SDXL on the left, FLUX on the right:

![Знімок екрана 2024-08-06 145710](https://github.com/user-attachments/assets/c550e47f-0368-4530-bc47-8ce934de6ff3)

From the above we can conclude that FLUX is great in artistic photos, but still very far from artistic illustration...

**Comment by wisdombox44** [comment]
only works at ComfyUI and SwarmUI at the moment if you don't like nodes try Swarm UI
here are online alternatives.
https://replicate.com/black-forest-labs
https://huggingface.co/spaces/black-forest-labs/FLUX.1-dev
https://huggingface.co/spaces/black-forest-labs/FLUX.1-schnell
https://tost.ai/
if you dont have strong PC better to try online services 
Have fun:)

**Reply by vicento** [reply]
does SwarmUI work on macs too ? i was not able to make ComfyUI work

**Reply by wisdombox44** [reply]
Yes it does
https://github.com/mcmonkeyprojects/SwarmUI
https://discord.com/invite/MYedrxEc
Join their discord server, the developer and community really helpful

**Comment by bsdesigner** [comment]
Also depressing about FLUX models is the rigid and unreasonable censorship that treats every second word as NSFW content. Here is a typical example from [replicate.com](https://replicate.com/black-forest-labs/flux-dev):

![Знімок екрана 2024-08-07 142317](https://github.com/user-attachments/assets/5f4196b2-bf38-40a1-a4f5-412fdbc84e98)

**Reply by mashb1t** [reply]
@bsdesigner While it is likely that nude and (female) celebrity images have not been used in the Flux training data, this specific case is a limitation of replicate, not the model itself. Models do not have nsfw checkers included as they are a separate piece of software, similar to the Fooocus NSFW checker.

**Comment by Moorer624** [comment]
You shouldn’t be starting something like this if you can’t continue to
support it. And now I’ll be deleting Fooocus.

**Reply by mashb1t** [reply]
=> https://github.com/lllyasviel/Fooocus/discussions/3504

**Reply by IPv6** [reply]
> You shouldn’t be starting something like this if you can’t continue to support it. And now I’ll be deleting Fooocus.

No, you should not give such "advices" to people doing stuff in free time, without pay :)
Because no one owe you nothing here, you know

**Comment by Moorer624** [comment]


**Reply by bsdesigner** [reply]
Thank you for letting us all know. I don't know what we'd do without this information )))

**Comment by CantainPike** [comment]
Anger does not fix problems. May learn coding and build your own webui.
